Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,469 @@
+                });
+            }
+            return list;
+        }
+
+
+        /// <summary>
+        /// Get Cash Balances from exchange
+        /// </summary>
+        /// <returns></returns>
+        //todo: handle other currencies
+        public override List<Securities.Cash> GetCashBalance()
+        {
+            var list = new List<Securities.Cash>();
+            var response = RestClient.GetBalances();
+            foreach (var item in response)
+            {
+                if (item.Type == wallet)
+                {
+                    if (item.Currency == "usd")
+                    {
+                        list.Add(new Securities.Cash(item.Currency, item.Amount, 1));
+                    }
+                    else
+                    {
+                        var ticker = RestClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                        list.Add(new Securities.Cash("BTC", item.Amount * divisor, decimal.Parse(ticker.Mid) / divisor));
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Get queued tick data
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Data.BaseData> GetNextTicks()
+        {
+            lock (ticks)
+            {
+                var copy = ticks.ToArray();
+                ticks.Clear();
+                return copy;
+            }
+        }
+
+        /// <summary>
+        /// Begin ticker polling
+        /// </summary>
+        /// <param name="job"></param>
+        /// <param name="symbols"></param>
+        public virtual void Subscribe(Packets.LiveNodePacket job, IEnumerable<Symbol> symbols)
+        {
+            var task = Task.Run(() => { this.RequestTicker(); }, _tickerToken.Token);
+        }
+
+        private void RequestTicker()
+        {
+            var response = RestClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+            lock (ticks)
+            {
+                ticks.Add(new Tick
+                {
+                    AskPrice = decimal.Parse(response.Ask) / divisor,
+                    BidPrice = decimal.Parse(response.Bid) / divisor,
+                    Time = Time.UnixTimeStampToDateTime(double.Parse(response.Timestamp)),
+                    Value = decimal.Parse(response.LastPrice) / divisor,
+                    TickType = TickType.Quote,
+                    Symbol = symbol,
+                    DataType = MarketDataType.Tick,
+                    Quantity = (int)(Math.Round(decimal.Parse(response.Volume), 2) * divisor)
+                });
+            }
+            if (!_tickerToken.IsCancellationRequested)
+            {
+                Thread.Sleep(8000);
```

Comment:
Many (hopefully all!) of the `Thread.Sleep` calls in the solution are on dedicated threads.